File: frontend/src/api/encyclopedia.ts
import { apiClient } from './client';
import type { ApiResponse } from '@/types/api';

// ============================================================
// Types
// ============================================================

export interface BreedFeatures {
  origin: string;
  size: string;
  weight: string;
  lifespan: string;
  coat: string;
  colors: string[];
}

export interface BreedCare {
  exercise: string;
  grooming: string;
  diet: string;
  training: string;
}

export interface BreedSummary {
  id: string;
  name: string;
  english_name: string;
  species: string;
  summary: string;
  image_emoji: string;
  popularity: number;
}

export interface BreedDetail extends BreedSummary {
  category: string;
  description: string;
  features: BreedFeatures;
  personality: string[];
  care_requirements: BreedCare;
  health_issues: string[];
  suitable_for: string[];
}

export interface HealthConditionSummary {
  id: string;
  name: string;
  species: string;
  category: string;
  severity: string;
  image_emoji: string;
}

export interface HealthCondition extends HealthConditionSummary {
  description: string;
  symptoms: string[];
  urgent_symptoms: string[];
  possible_causes: string[];
  treatment: string[];
  home_care: string[];
  prevention: string[];
}

export interface HealthCategoryGroup {
  category: string;
  category_label: string;
  conditions: HealthConditionSummary[];
}

export const SEVERITY_CONFIG: Record<string, { label: string; variant: 'safe' | 'warning' | 'danger' | 'emergency' }> = {
  mild: { label: '轻微', variant: 'safe' },
  moderate: { label: '中等', variant: 'warning' },
  severe: { label: '严重', variant: 'danger' },
  emergency: { label: '紧急', variant: 'emergency' },
};

export const CATEGORY_CONFIG: Record<string, { label: string; emoji: string }> = {
  digestive: { label: '消化系统', emoji: '🍽️' },
  respiratory: { label: '呼吸系统', emoji: '🫁' },
  skin: { label: '皮肤系统', emoji: '🧴' },
  urinary: { label: '泌尿系统', emoji: '💧' },
  eye: { label: '眼部疾病', emoji: '👁️' },
  dental: { label: '口腔疾病', emoji: '🦷' },
  parasite: { label: '寄生虫', emoji: '🪱' },
  infectious: { label: '传染病', emoji: '🦠' },
  skeletal: { label: '骨骼关节', emoji: '🦴' },
};

// ============================================================
// API Functions
// ============================================================

export const encyclopediaApi = {
  getBreeds: async (species: string): Promise<BreedSummary[]> => {
    const res = await apiClient.get<ApiResponse<{ species: string; breeds: BreedSummary[] }>>(
      `/api/v1/encyclopedia/breeds/${species}`
    );
    return res.data?.breeds || [];
  },

  getBreedDetail: async (breedId: string): Promise<BreedDetail | null> => {
    const res = await apiClient.get<ApiResponse<{ breed: BreedDetail }>>(
      `/api/v1/encyclopedia/breed/${breedId}`
    );
    return res.data?.breed || null;
  },

  getHealthConditions: async (species: string): Promise<HealthCategoryGroup[]> => {
    const res = await apiClient.get<ApiResponse<{ species: string; categories: HealthCategoryGroup[] }>>(
      `/api/v1/encyclopedia/health/${species}`
    );
    return res.data?.categories || [];
  },

  getHealthDetail: async (conditionId: string): Promise<HealthCondition | null> => {
    const res = await apiClient.get<ApiResponse<{ condition: HealthCondition }>>(
      `/api/v1/encyclopedia/health/detail/${conditionId}`
    );
    return res.data?.condition || null;
  },

  searchKnowledge: async (query: string): Promise<{ breeds: BreedSummary[]; conditions: HealthConditionSummary[] }> => {
    const res = await apiClient.get<ApiResponse<{ breeds: BreedSummary[]; conditions: HealthConditionSummary[] }>>(
      `/api/v1/encyclopedia/search`,
      { query }
    );
    return res.data || { breeds: [], conditions: [] };
  },
};

File: animal/src/api/schemas/encyclopedia.py
from typing import List, Optional
from pydantic import BaseModel, Field
from datetime import datetime


class BreedFeatures(BaseModel):
    origin: str = Field("", description="原产地")
    size: str = Field("", description="体型大小")
    weight: str = Field("", description="体重范围")
    lifespan: str = Field("", description="平均寿命")
    coat: str = Field("", description="被毛类型")
    colors: List[str] = Field(default_factory=list, description="毛色")


class BreedCare(BaseModel):
    exercise: str = Field("", description="运动需求")
    grooming: str = Field("", description="美容护理")
    diet: str = Field("", description="饮食建议")
    training: str = Field("", description="训练难度")


class BreedDetail(BaseModel):
    id: str = Field(..., description="品种ID")
    name: str = Field(..., description="品种名称")
    english_name: str = Field("", description="英文名称")
    species: str = Field(..., description="物种: cat/dog")
    category: str = Field("", description="品种类别")
    description: str = Field("", description="品种简介")
    summary: str = Field("", description="一句话概述")
    features: BreedFeatures = Field(default_factory=BreedFeatures, description="标准特征")
    personality: List[str] = Field(default_factory=list, description="性格特点")
    care_requirements: BreedCare = Field(default_factory=BreedCare, description="饲养要求")
    health_issues: List[str] = Field(default_factory=list, description="常见健康问题")
    suitable_for: List[str] = Field(default_factory=list, description="适合人群")
    image_emoji: str = Field("🐾", description="品种表情符号")
    popularity: int = Field(5, ge=1, le=10, description="受欢迎程度 1-10")


class BreedSummary(BaseModel):
    id: str = Field(..., description="品种ID")
    name: str = Field(..., description="品种名称")
    english_name: str = Field("", description="英文名称")
    species: str = Field(..., description="物种: cat/dog")
    summary: str = Field("", description="一句话概述")
    image_emoji: str = Field("🐾", description="品种表情符号")
    popularity: int = Field(5, ge=1, le=10, description="受欢迎程度")


class BreedListResponse(BaseModel):
    species: str = Field(..., description="物种: cat/dog")
    breeds: List[BreedSummary] = Field(..., description="品种列表")


class BreedDetailResponse(BaseModel):
    breed: BreedDetail = Field(..., description="品种详细信息")


class HealthCondition(BaseModel):
    id: str = Field(..., description="病症ID")
    name: str = Field(..., description="病症名称")
    species: str = Field(..., description="适用物种: cat/dog/both")
    category: str = Field(..., description="病症类别")
    description: str = Field("", description="病症描述")
    symptoms: List[str] = Field(default_factory=list, description="常见症状")
    urgent_symptoms: List[str] = Field(default_factory=list, description="需立即就医的症状")
    possible_causes: List[str] = Field(default_factory=list, description="可能病因")
    severity: str = Field("mild", description="严重程度: mild/moderate/severe/emergency")
    treatment: List[str] = Field(default_factory=list, description="治疗方案")
    home_care: List[str] = Field(default_factory=list, description="日常护理建议")
    prevention: List[str] = Field(default_factory=list, description="预防措施")
    image_emoji: str = Field("🏥", description="表情符号")


class HealthConditionSummary(BaseModel):
    id: str = Field(..., description="病症ID")
    name: str = Field(..., description="病症名称")
    species: str = Field(..., description="适用物种")
    category: str = Field(..., description="病症类别")
    severity: str = Field(..., description="严重程度")
    image_emoji: str = Field("🏥", description="表情符号")


class HealthCategoryGroup(BaseModel):
    category: str = Field(..., description="疾病类别")
    category_label: str = Field("", description="类别名称")
    conditions: List[HealthConditionSummary] = Field(..., description="病症列表")


class HealthListResponse(BaseModel):
    species: str = Field(..., description="物种: cat/dog/both")
    categories: List[HealthCategoryGroup] = Field(..., description="按类别分组的病症列表")


class HealthDetailResponse(BaseModel):
    condition: HealthCondition = Field(..., description="病症详细信息")


class KnowledgeCategoriesResponse(BaseModel):
    categories: List[dict] = Field(..., description="知识分类列表")


class SearchKnowledgeResponse(BaseModel):
    breeds: List[BreedSummary] = Field(default_factory=list, description="匹配的品种")
    conditions: List[HealthConditionSummary] = Field(default_factory=list, description="匹配的病症")
